python/django/security/user-open.yaml

```diff
@@ -0,0 +1,66 @@
+rules:
+  - id: user-open
+    patterns:
+      - pattern-inside: |
+          def $F(...):
+            ...
+      - pattern-either:
+          - pattern: open(..., request.$W.get(...), ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              open(..., $V, ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              $A = open(..., $V, ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              return open(..., $V, ...)
+          - pattern: open(..., request.$W(...), ...)
```

Comment:
What is this pattern getting at? Looks like it could be `request.GET(...)`? Maybe this is too general, if there's specific function calls we're concerned about we should list them.

Reply:
There are a couple of methods on the request objects that are conveniences for getting certain attributes, as an example: `request.get_host()`

cf. https://docs.djangoproject.com/en/3.0/ref/request-response/#django.http.HttpRequest.get_host